src/parser.js

```diff
@@ -777,7 +784,13 @@ export default class Parser extends Stream {
   end() {
     // flush any buffered input
     this.lineStream.push('\n');
+    if (this.manifest.dateRanges && this.lastProgramDateTime === null) {
```

Comment:
Maybe it makes sense to add dateRanges as the default array in the constructor:

```
    this.manifest = {
      allowCache: true,
      discontinuityStarts: [],
      segments: [],
      dateRanges: [],
    };
```

And here, we should check `this.manifest.dateRanges.length && ...`

we also can remove this line, in this case:
`this.manifest.dateRanges = this.manifest.dateRanges || [];`